java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCStoGCS.java

```diff
@@ -0,0 +1,176 @@
+
+  private String tempQuery;
+
+  public GCStoGCS() {
+
+    projectID = getProperties().getProperty(PROJECT_ID_PROP);
+    inputFileLocation = getProperties().getProperty(GCS_GCS_INPUT_LOCATION);
+    inputFileFormat = getProperties().getProperty(GCS_GCS_INPUT_FORMAT);
+    gcsOutputLocation = getProperties().getProperty(GCS_GCS_OUTPUT_LOCATION);
+    gcsPartitionColumn = getProperties().getProperty(GCS_GCS_OUTPUT_PARTITION_COLUMN);
+    gcsOutputFormat = getProperties().getProperty(GCS_GCS_OUTPUT_FORMAT);
+    gcsWriteMode = getProperties().getProperty(GCS_GCS_WRITE_MODE);
+    tempTable = getProperties().getProperty(GCS_GCS_TEMP_TABLE);
+    tempQuery = getProperties().getProperty(GCS_GCS_TEMP_QUERY);
+  }
+
+  @Override
+  public void runTemplate() {
+    validateInput();
+
+    SparkSession spark = null;
+    LOGGER.info(
+        "Starting GCS to GCS spark job with following parameters:"
+            + "1. {}:{}"
+            + "2. {}:{}"
+            + "3. {}:{}"
+            + "4. {}:{}"
+            + "5. {}:{}",
+        GCS_GCS_INPUT_LOCATION,
+        inputFileLocation,
+        GCS_GCS_INPUT_FORMAT,
+        inputFileFormat,
+        GCS_GCS_OUTPUT_LOCATION,
+        gcsOutputLocation,
+        GCS_GCS_OUTPUT_FORMAT,
+        gcsOutputFormat,
+        GCS_GCS_WRITE_MODE,
+        gcsWriteMode);
+
+    try {
+      spark = SparkSession.builder().appName("GCS to GCS load").getOrCreate();
+
+      Dataset<Row> inputData = null;
+
+      switch (inputFileFormat) {
+        case GCS_GCS_CSV_FORMAT:
+          inputData =
+              spark
+                  .read()
+                  .format(GCS_GCS_CSV_FORMAT)
+                  .option(GCS_GCS_CSV_HEADER, true)
+                  .option(GCS_GCS_CSV_INFOR_SCHEMA, true)
+                  .load(inputFileLocation);
+          break;
+        case GCS_GCS_AVRO_FORMAT:
+          inputData = spark.read().format(GCS_GCS_AVRO_EXTD_FORMAT).load(inputFileLocation);
+          break;
+        case GCS_GCS_PRQT_FORMAT:
+          inputData = spark.read().parquet(inputFileLocation);
+          break;
+        default:
+          throw new IllegalArgumentException(
+              "Currently avro, parquet and csv are the only supported formats");
+      }
+
+      if (tempTable != null && tempQuery != null) {
+        inputData.createOrReplaceGlobalTempView(tempTable);
+        inputData = spark.sql(tempQuery);
+      }
+
+      inputData.show();
+
+      DataFrameWriter<Row> writer = inputData.write().mode(gcsWriteMode).format(gcsOutputFormat);
+
+      /*
+       * If optional partition column is passed than partition data by partition
+       * column before writing to GCS.
+       * */
+      if (StringUtils.isNotBlank(gcsPartitionColumn)) {
+        LOGGER.info("Partitioning data by :{} cols", gcsPartitionColumn);
+        writer = writer.partitionBy(gcsPartitionColumn);
+      }
+
+      spark.conf().set("mapreduce.fileoutputcommitter.marksuccessfuljobs", "false");
```

Comment:
why is this property needed ?
If it is really needed, please put a code comment to explain the same.

Reply:
Actually whenever, we perform a transformation and puts the data back in the form of a file. Spark by default creates a success file inside the target folder/directory. Hence, this command will make that default property as false. So that, spark should not create any success file in target folder/directory.

Reply:
Added comment as per suggestion.